src/libostree/ostree-repo.c

```diff
@@ -1266,6 +1266,86 @@ impl_repo_remote_merge (OstreeRepo     *self,
   return TRUE;
 }
 
+static gboolean
+impl_repo_remote_replace (OstreeRepo     *self,
+                          GFile          *sysroot,
+                          const char     *name,
+                          const char     *url,
+                          GVariant       *options,
+                          GCancellable   *cancellable,
+                          GError        **error)
+{
+  g_return_val_if_fail (name != NULL, FALSE);
+  g_return_val_if_fail (url != NULL, FALSE);
+  g_return_val_if_fail (options == NULL || g_variant_is_of_type (options, G_VARIANT_TYPE ("a{sv}")), FALSE);
+
+  if (!ostree_validate_remote_name (name, error))
+    return FALSE;
+
+  g_autoptr(GError) local_error = NULL;
+  g_autoptr(OstreeRemote) remote = _ostree_repo_get_remote (self, name, &local_error);
+  if (remote == NULL)
+    {
+      if (!g_error_matches (local_error, G_IO_ERROR, G_IO_ERROR_NOT_FOUND))
+        {
+          g_propagate_error (error, g_steal_pointer (&local_error));
+          return FALSE;
+        }
+      if (!impl_repo_remote_add (self, sysroot, FALSE, name, url, options,
+                                 cancellable, error))
+        return FALSE;
+    }
+  else
+    {
+      /* Replace the entire option group */
+      if (!g_key_file_remove_group (remote->options, remote->group, error))
+        return FALSE;
+
+      if (g_str_has_prefix (url, "metalink="))
+        g_key_file_set_string (remote->options, remote->group, "metalink",
+                               url + strlen ("metalink="));
+      else
+        g_key_file_set_string (remote->options, remote->group, "url", url);
```

Comment:
Hm, at this point we should extract this to a helper I'd say.

Reply:
Right, I didn't spend a lot of time thinking about refactoring things. I mostly wanted to get the feature and tests in place to see what you thought. But I'll look through again and see if there are clear winners.